nexus/reconfigurator/planning/src/blueprint_builder.rs

```diff
@@ -767,15 +773,18 @@ impl BlueprintBuilderRng {
 /// that we've changed and a _reference_ to the parent blueprint's zones.  This
 /// struct makes it easy for callers iterate over the right set of zones.
 struct BlueprintZonesBuilder<'a> {
-    changed_zones: BTreeMap<Uuid, BlueprintZonesConfig>,
-    parent_zones: &'a BTreeMap<Uuid, BlueprintZonesConfig>,
+    changed_zones: BTreeMap<TypedUuid<SledKind>, BlueprintZonesConfig>,
+    // Temporarily make a clone of the parent blueprint's zones so we can use
+    // typed UUIDs everywhere. Once we're done migrating, this `Cow` can be
+    // removed.
+    parent_zones: Cow<'a, BTreeMap<TypedUuid<SledKind>, BlueprintZonesConfig>>,
```

Comment:
Just making sure I understand: the `Cow` is technically unnecessary (we always construct it as `Cow::Owned`), but is just here as a hint to remind us to switch it back to `&'a BTreeMap<...>`?